### PR TITLE
impl for  Eip2930

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
ref : https://github.com/paradigmxyz/reth/issues/4908

### CODE REVIEW

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 1]
```diff
@@ -302,6 +322,7 @@ impl MockTransaction {
             MockTransaction::Legacy { gas_price, .. } => *gas_price,
             MockTransaction::Eip1559 { max_fee_per_gas, .. } => *max_fee_per_gas,
             MockTransaction::Eip4844 { max_fee_per_gas, .. } => *max_fee_per_gas,
+            MockTransaction::Eip2930 { .. } => 0,
```
Comment (mattsse) [inline crates/transaction-pool/src/test_utils/mock.rs:325]:
```suggestion
            MockTransaction::Eip2930 { gas_price,.. } => *gas_price
```

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 2]
```diff
@@ -429,6 +454,7 @@ impl PoolTransaction for MockTransaction {
             MockTransaction::Eip4844 { max_fee_per_gas, value, gas_limit, .. } => {
                 U256::from(*gas_limit) * U256::from(*max_fee_per_gas) + *value
             }
+            MockTransaction::Eip2930 { .. } => U256::ZERO,
```
Comment (mattsse) [inline crates/transaction-pool/src/test_utils/mock.rs:457]: needs same calc as ::Legacy

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 3]
```diff
@@ -441,6 +467,7 @@ impl PoolTransaction for MockTransaction {
             MockTransaction::Legacy { gas_price, .. } => *gas_price,
             MockTransaction::Eip1559 { max_fee_per_gas, .. } => *max_fee_per_gas,
             MockTransaction::Eip4844 { max_fee_per_gas, .. } => *max_fee_per_gas,
+            MockTransaction::Eip2930 { .. } => 0,
```
Comment (mattsse) [inline crates/transaction-pool/src/test_utils/mock.rs:470]:
```suggestion
            MockTransaction::Eip2930 { gas_price,.. } => *gas_price,
```

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 4]
```diff
@@ -461,6 +489,7 @@ impl PoolTransaction for MockTransaction {
             MockTransaction::Eip4844 { max_priority_fee_per_gas, .. } => {
                 Some(*max_priority_fee_per_gas)
             }
+            MockTransaction::Eip2930 { .. } => Some(0),
```
Comment (mattsse) [inline crates/transaction-pool/src/test_utils/mock.rs:492]:
```suggestion
            MockTransaction::Eip2930 { .. } => None,
```

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 5]
```diff
@@ -491,6 +520,7 @@ impl PoolTransaction for MockTransaction {
             MockTransaction::Legacy { gas_price, .. } => *gas_price,
             MockTransaction::Eip1559 { max_priority_fee_per_gas, .. } => *max_priority_fee_per_gas,
             MockTransaction::Eip4844 { max_priority_fee_per_gas, .. } => *max_priority_fee_per_gas,
+            MockTransaction::Eip2930 { .. } => 0,
```
Comment (mattsse) [inline crates/transaction-pool/src/test_utils/mock.rs:523]:
```suggestion
            MockTransaction::Eip2930 { gas_price,.. } => *gas_price,
```

**Review by mattsse — CHANGES_REQUESTED** [review]
great, a few nits

also needs nightly formatting `cargo +nightly fmt`

**Comment by DoTheBestToGetTheBest** [comment]
> great, a few nits
> 
> also needs nightly formatting `cargo +nightly fmt`

Thank you so so much Mat. I really appreciate that. I'm really sorry i don't have actually nightly installed in my PC

**Review by mattsse — APPROVED** [review]
nice, this is great

**Comment by DoTheBestToGetTheBest** [comment]
> nice, this is great

Thank you so so much Mat. You're amazing !

**Pull request merged**